pkg/apis/batch/types.go

```diff
@@ -138,6 +138,18 @@ type JobSpec struct {
 
 	// Describes the pod that will be created when executing a job.
 	Template api.PodTemplateSpec
+
+	// ttlSecondsAfterFinished limits the lifetime of a Job that has finished
+	// execution (either Complete or Failed). If this field is set, once the Job
+	// finishes, it will be deleted after ttlSecondsAfterFinished expires. When
+	// the Job is being deleted, its lifecycle guarantees (e.g. finalizers) will
+	// be honored. If this field is unset, ttlSecondsAfterFinished will not
+	// expire. If this field is set to zero, ttlSecondsAfterFinished expires
+	// immediately after the Job finishes.
+	// This field is alpha-level and is only honored by servers that enable the
+	// TTLAfterFinished feature.
+	// +optional
+	TTLSecondsAfterFinished *int32
```

Comment:
status uses `CompletionTime`, should this be `TTLAfterCompletionSeconds`?

Reply:
or is CompletionTime not set in failure cases?

Reply:
CompletionTime isn't set in failure cases.